### PR TITLE
feat(viewport-panes): add viewport-pane component

### PR DESCRIPTION
**Pull request by jefflembeck**
This gives us the ability to add a class to the top pane of the layout
and have it scale based on the size of the viewport's height.

[Finishes: #113681439]

BREAKING CHANGE: Adds variables that will affect the header as well

### CODE REVIEW

File: src/pivotal-ui/components/pui-variables.scss [hunk 1]
```diff
@@ -1326,4 +1326,7 @@ $header-subtle-text-color: rgba(255,255,255,0.8);
 $header-nav-highlight-bg: #a42b2b;
 $desktopMinWidth: $screen-sm;
 $greigh: rgba(195,195,195,1); // median greigh
-$greigh3: rgba(black, 0.6);
\ No newline at end of file
+$greigh3: rgba(black, 0.6);
+
+$header-height-max-sm: 7.478125em;
```
Comment (stubbornella) [inline src/pivotal-ui/components/pui-variables.scss:1334]: hehe. :)

**Comment by jefflembeck** [comment]
There are some bizarre numbers in here. They come from the great land of measurement of how tall the header is.

This is unfortunate, as there are small bugs in the header which will undoubtedly affect its size.

We can update things in order to fix this, and likely will have to, now that I've set height restrictions on the header itself, but coming up with an ideal solution gets in the way of having a solution that works right now, and this does that.

File: src/pivotal-ui/components/viewport-panes/viewport-panes.scss [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  <div class="container bg-glow">
+    <h1 class="type-neutral-11">This is a pane</h1>
+  </div>
+</div>
+```
+
+*/
+
+.viewport-pane {
+  height: calc(100vh - $header-height-max-sm);
```
Comment (stubbornella) [inline src/pivotal-ui/components/viewport-panes/viewport-panes.scss]: clever monkey! :monkey: love it

File: src/pivotal-ui/components/viewport-panes/viewport-panes.scss [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+.hero-transparent + .viewport-pane {
+  box-sizing: content-box;
+}
+
+@media (min-width: $screen-sm-min) {
+  .viewport-pane {
+    height: calc(80vh - $header-height-max-lg);
```
Comment (stubbornella) [inline src/pivotal-ui/components/viewport-panes/viewport-panes.scss]: I think the math may be slightly off here. It needs to leave room for both the header and the enterprise pane below it (the word "enterprise" in the next pane needs to show).

Actually wait, this is perfect for the one on the On-Site landing page. We just need an additional one for the homepage.

**Comment by stubbornella** [comment]
This is rad and will work perfectly (I think, though I didn't pull it and test it) on the on-site landing page, we just need to make one that isn't quite so tall and leaves room to see the enterprise secondary hero on the homepage.

**Comment by stubbornella** [comment]
Can you upload a screen grab of the result that includes your browser chrome?

File: src/pivotal-ui/components/viewport-panes/viewport-panes.scss (hunk 3 above)
Comment (stubbornella) [inline src/pivotal-ui/components/viewport-panes/viewport-panes.scss]: also, I think this should be min-height. In the unfortunate case where we have too much content for on screen height, we don't want things to get cropped.

File: src/pivotal-ui/components/pui-variables.scss (hunk 1 above)
Comment (aredridel) [inline src/pivotal-ui/components/pui-variables.scss:1334]: Oh god these magic numbers!

File: src/pivotal-ui/components/viewport-panes/viewport-panes.scss (hunk 3 above)
Comment (aredridel) [inline src/pivotal-ui/components/viewport-panes/viewport-panes.scss]: Ooh good call

**Comment by jefflembeck** [comment]
bring-the-pane

**Comment by aredridel** [comment]
Can confirm: pane both paney and viewporty: 
screen shot 2016-03-11 at 16 41 47

**Pull request merged**